stratum/hal/lib/common/common.proto

```diff
@@ -679,9 +679,20 @@ message TofinoConfig {
     map<uint32, BfPerPortShapingConfig> per_port_shaping_configs = 1;
   }
 
+  // DeflectOnPacketDropConfig specifies the port and queue where dropped
+  // packets will be deflected to.
+  message DeflectOnPacketDropConfig {
+    message DropTarget {
+      uint32 port = 1;
```

Comment:
We might need to change that to `oneof { port, sdk_port}`, else we have to add the recirc ports to the singleton port list.

Reply:
I think we can use `oneof`. I don't think we cannot add recirculate port to the singleton port list since we cannot find the SDK port number for the recirculate port by using port/channel

Reply:
should we also add `pipe` to this config for specific pipes?

Reply:
I see some inconsistency with port shaping config where the port id is passed as a map key. Not a big deal, since this is an internal API, but I was wondering if you also see this as a potential source of confusion and if you have given some thought to how to make the two configs similar. For example, should we use the `oneof` approach for both port shaping and dod?

Reply:
Notes from sync:

- `oneof` requires a flat config structure with repeated fields instead of nested maps
- might want to change port shaping from nested maps to flat (future PR)